Aesop/Util/Basic.lean

```diff
@@ -227,7 +223,7 @@ def deduplicateSorted [eq : BEq α] (xs : Array α) : Array α :=
   xs.mergeAdjacentDuplicates (λ x _ => x)
 
 set_option linter.unusedVariables false in
-def deduplicate [Inhabited α] [ord : Ord α] (xs : Array α) : Array α :=
+def deduplicate [Inhabited α] [BEq α] [ord : Ord α] (xs : Array α) : Array α :=
```

Comment:
IMHO, this function should be renamed or removed.  `Array.deduplicate` is prime naming estate, and this function doesn't live up to the name: it also sorts the array in addition to deduplicating it!

(It's also unused in aesop, as far as I can tell.)

Reply:
I have open PRs which move this entire file into std4. Once these are merged, Aesop will only use the Aesop namespace.